frontend/src/deviceDetailsPage/DeviceNameDetail.jsx

```diff
@@ -0,0 +1,117 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Box, Button, TextField } from "@material-ui/core";
+import EditIcon from "@material-ui/icons/Edit";
+import * as DeviceApi from "../api/DeviceApi";
+
+export default class DeviceNameDetail extends React.Component {
+  constructor(props) {
+    super(props);
+    this.deviceName = props.deviceName;
+    this.deviceId = props.deviceId;
+    this.state = {
+      editing: false,
+      name: this.deviceName
+    };
+
+    this.startEdit = this.startEdit.bind(this);
+    this.cancelEditing = this.cancelEditing.bind(this);
+    this.confirmEditing = this.confirmEditing.bind(this);
+    this.setName = this.setName.bind(this);
+  }
+
+  setName(updatedName) {
+    this.setState({ name: updatedName });
+  }
+
+  startEdit() {
+    this.setState({ editing: true });
+  }
+
+  cancelEditing() {
+    this.setState({ name: this.deviceName });
```

Comment:
Why is this setState for name necessary when you aren't changing the name of the device? Is it because of the onChange to the text field target value it temporary sets the name state to the value in the text field but if you cancel you want to reset it to the original device name before editing was initiated?

Reply:
more or less yeah! since we don't have refresh happening on the page but still want to show the updated name after changes have been saved, we use the name saved in the component's state for display 
that meant that if you made a change and then pressed cancel, the device name would end up displaying the changes you meant to discard, so i added this line :)